### ISSUE TITLE
WeGlide Copilot Data implementation

### ISSUE THREAD

**Issue by XCNav** [XCSoar/XCSoar#1403]
XCSoar version
--------------

7.42 Android 11/13

What should XCSoar do differently, what functionality should be added?
----------------------------------------------------------------------

As there already exists a connection to the WeGlide servers, it would be perfect to add the WeGlide Copilot Data into the MapDisplay.

What are you trying to do, what is the use case for the suggestion?
-------------------------------------------------------------------
In the WeGlide App you can choose different actual information layers as well based on OGN/Flarm data if you have a subscription. As you can already log on the WeGlide server, this should be accessible cvia XCS.
That would make the use of a second phone or switching between XCS and WeGlide App obsolete.

**Comment by lordfolken** [comment]
Is this data accessible via api? 
Do you have documentation for this? 
Do you have someone that can implement this?

**Comment by XCNav** [comment]
No, not yet. I addressed this to one of the  WeGlide guys with a link to this issue.

**Comment by XCNav** [comment]
Not yet, please keep it open, will work on it this week!

**Comment by LeNinjaHD** [comment]
Is there any news on this?

**Comment by XCNav** [comment]
> @XCNav have you contact weglide regarding this?

I contacted Johannes from We glide with my concern but got no response. Just started a second approach!

**Comment by LeNinjaHD** [comment]
Alright, i got a response from WeGlide! They are willing to provide their APIs if a XCSoar Maintainer contacts them. They told me however that they are a bit sceptical, beacause they provided API in the past which never made it into the App.

**Comment by LeNinjaHD** [comment]
> I just want you to realize what you are asking me to do here: This is the ai's estimation of this kind of effort:
> 
> client-side estimate
> 
> If backend provides simple HTTP tiles/grids (recommended v1)
> ~2–4 weeks of focused work
> Network client + scheduler (net/client/... style like tim): 2–4 days
> Cache + expiry + retry/backoff: 2–4 days
> Map overlay renderer integration (MapWindow layer, zoom/opacity): 3–6 days
> Settings UI (enable/disable, opacity, legend, age display): 2–4 days
> Cross-backend rendering fixes (OpenGL + memory canvas + GDI): 2–5 days
> Testing/polish (offline behavior, stale data, performance): 3–5 days

I thought WeGlide might be interested in helping implementing this, however I have heard that they are working on their own app at the moment, so unfortunately this wont happen I believe.

**Comment by yorickreum** [comment]
> They want to talk to a XCSoar maintainer first about the scope.

Are they aware of this issue here? Would be a good place to discuss the scope imo.

**Comment by lordfolken** [comment]
I wrote them an email. Lets see.
